src/shared/tpm2-util.c

```diff
@@ -1306,3 +1374,12 @@ int tpm2_primary_alg_from_string(const char *alg) {
                 return TPM2_ALG_RSA;
         return -EINVAL;
 }
+
+bool tpm2_valid_pin(const char *pin) {
+        assert(pin);
+
+        /* Enforce a PIN length of at least 4 characters: Fewer characters provide little protection, even
+         * with dictionary attack (DA) protection. In the unlock case, validation helps to avoid unnecessary
+         * unlock attempts that increment the DA counter. */
+        return strlen(pin) >= 4;
+}
```

Comment:
so i am not sure we should do this kind of validation here, but rather defer to libpwquality on the client side really. i.e. we validate passwords anyway client side, and I am not sure we should validate PINs in any different way (after all it's just a different name for the exact same thing...). 

(I mean, it might be OK to use slightly weaker settings for libpwquality when setting things up with a TPM2 backing, given the DA protection, but the point I am trying to make here is that we shouldn't have two layers of pw quality checks: a simplistic checking here — which ends up being placed server-side – and a complex, detailed checking client-side, it should use the same code/logic, and be placed at the same possitions, and at most differ in selected validation levels)

Reply:
What's client side and server side? Are you talking about password agents?

Anyway, since this is a bit controversial, I'll just remove the length check completely.

Reply:
Forget client/server side, doesn't matter.